### PR TITLE
check the outputs for cellcount and age prediction

### LINKED ISSUE #20: [Code Refactor]: Enhance the ./check_upload.sh script for checking the cor_plot.pdf
### Contact Details

s.w.wang@exeter.ac.uk

### Path to affected file

```shell
`./check_upload.sh 03a check`
```

### Type of refactor

Edge case error handling

### Description of required code refactor

Since my DNA methylation data were generated from the whole blood, I set `cellcounts_required="yes" and measured_cellcounts="NULL"` in my config file. This configuration causes the 03a script to skip the step of "Comparing measured with predicted cell counts," and as a result, no `/results/03/cor_plot.pdf` is generated.

However, the `./check_upload.sh` script currently checks for the existence of `/results/03/cor_plot.pdf` regardless of the dataset's specific configuration. Consequently, the absence of this file leads to a failure when running `./check_upload.sh`.

To address this issue, I suggest enhancing the `./check_upload.sh` script by adding conditional checks for output files based on the specific configuration of the dataset.

Thank you.

### PR DESCRIPTION
**Pull request by SiyiSEA**
This pull request is for:

1. Enhance the function of `check_upload.sh 03a check` to handle the situation when `cellcounts_required="yes" and measured_cellcounts="NULL"` have been set in the config file;
2. Check all the outputs for 03a.

### CODE REVIEW

**Review by epzjlm — APPROVED** [review]

File: resources/logs/check_results.sh [hunk 1]
```diff
@@ -66,58 +66,92 @@ check_results_02 () {
...
+			echo "Plots of cell counts are present"
+		else
+			echo "Problem: plots of cell counts are absent"
+			exit 1
+		fi
+	else
+		echo "Message: since cellcounts_required is not required, no output for cellcounts_summary.txt and cellcounts_plot.pdf for cell counts."
+	fi
 
-    if [ "${cellcounts_required}" = "yes" ]; then
```
Comment (ejh243) [inline resources/logs/check_results.sh:93]: Aren't we forcing all cohorts to recalulcate the cell composition variables? So the check for the plot should just be based on whether the cohort has observed counts to plot against? i.e. if measured_cellcounts != "NULL"

**Comment by SiyiSEA** [comment]
If we did force the cell_count = "yes",

Would that be more safe for us to delete the parameter of `cellcounts_required` in the config file?
And update the wiki description about the cell count? https://github.com/genetics-of-dna-methylation-consortium/godmc_phase2/wiki/Install-and-set-up#cell-count-data

I could do them.

**Comment by ejh243** [comment]
Currently it seems like recalculation of cell counts is not enforced, i.e. if `cellcounts_required `is set to no, this will be skipped. Should this be the case @epzjlm ?

**Comment by ejh243** [comment]
Looking at the wiki I wonder if this is because sorted datasets don't need cell composition caluclated? But is this is the case can't we use the `sorted_methylation` config variable to turn this off? Otherwise we might get whole blood datasets not running this.

**Comment by epzjlm** [comment]
It might be easer to add a check to script 01 or script 03

**Comment by ejh243** [comment]
I have removed the `cellcounts_required` variable and instead made better use of the `sorted_methylation` and `measured_cellcounts` variables. IMO even for sorted datasets the cell composition variables generated from the data are useful QC things to have. @SiyiSEA could you review my edits carefully in case it doesn't make sense.

**Comment by ejh243** [comment]
If both @SiyiSEA and @epzjlm are then happy we can merge this change and I will remove reference to the `cellcounts_required` variable from the wiki.

**Review by SiyiSEA — COMMENTED** [review]
agree!

**Pull request merged**